src/server/editorServices.ts

```diff
@@ -1217,10 +1217,43 @@ namespace ts.server {
             }
             else {
                 const project = this.createProject(configFilename, projectOptions);
+                let programSizeForNonTsFiles = 0;
+
+                // As the project openning might not be complete if there are too many files,
+                // therefore to surface the diagnostics we need to make sure the given client file is opened.
+                if (clientFileName) {

```

Comment:
The file reading here doesn't have an upper limit, therefore it could be much more than 20M. In the repro case it is around 100M, and I noticed server crashes. So I kept the limit here as well.